CycloneDX/Services/ProjectAssetsFileService.cs

```diff
@@ -143,15 +144,24 @@ private static void ResolveDependencyVersionRanges(HashSet<DotnetDependency> run
                     {
                         var normalizedDependencyKey = dependency.Key.ToLowerInvariant();
                         var package = runtimePackagesLookup[normalizedDependencyKey].FirstOrDefault(pkg => versionRange.Satisfies(NuGetVersion.Parse(pkg.Version)));
-                        if (package != default)
-                        {
-                            runtimePackage.Dependencies[dependency.Key] = package.Version;
-                        }
-                        else
+                        string patchVersion = package?.Version;
+
+                        // This can happen for dev-only packages that only have SDK functionality.
+                        if (package == default)
                         {
-                            // This should not happen, since all dependencies are resolved to a specific version.
-                            Console.Error.WriteLine($"Dependency ({dependency.Key}) with version range ({dependency.Value}) referenced by (Name:{runtimePackage.Name} Version:{runtimePackage.Version}) did not resolve to a specific version.");
+                            // Central package versions can already limit the expected range.
+                            if (!centralPackageVersions.TryGetValue(dependency.Key, out var cpv))
+                            {
+                                // If we don't have a CPV, ignore for now.
+                                // It will be tried to match later in the Runner again.
+                                continue;
+                            }
+
+                            // Despite the name, this could already be a non-range version.
+                            patchVersion = cpv.VersionRange.ToShortString();
```

Comment:
`ToShortString` generated the simplest name to be used:
* 1 -> 1.0.0
* [1, ) -> 1.0.0
* (1, ) -> (1.0.0, )
* [1, 2) -> [1.0.0, 2.0.0)

Non-min-inclusive version ranges would not be resolvable by the old version of the code, but will be resolvable in the new run for recursive project or solution runs.